ttnn/cpp/ttnn/operations/moreh/moreh_clip_grad_norm/moreh_clip_grad_norm_step2/device/moreh_clip_grad_norm_step2_device_operation.cpp

```diff
@@ -0,0 +1,68 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "moreh_clip_grad_norm_step2_device_operation.hpp"
+
+#include "common/constants.hpp"
+#include "ttnn/operations/core/compute_kernel/compute_kernel_config.hpp"
+#include "ttnn/operations/moreh/moreh_helper_functions.hpp"
+#include "ttnn/tensor/tensor.hpp"
+
+namespace ttnn::operations::moreh::moreh_clip_grad_norm_step2 {
+
+void MorehClipGradNormStep2Operation::validate_inputs(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+    ttnn::operations::check_tensor(tensor_args.tmp_pow_sum, "moreh_clip_grad_norm_step2", "tmp_pow_sum");
+
+    if (tensor_args.total_norm.has_value())
+        ttnn::operations::check_tensor(tensor_args.total_norm, "moreh_clip_grad_norm_step2", "total_norm");
```

Comment:
```suggestion
        check_tensor(tensor_args.total_norm, "moreh_clip_grad_norm_step2", "total_norm");
```

Reply:
Resolved in [06f7fe7](https://github.com/tenstorrent/tt-metal/pull/14667/commits/06f7fe78b8714d808e3442033a18a39ce88500c8)